joinmarket: 0.8.2 -> 0.8.3

nixbitcoin opened the pull request:
Update JoinMarket. See commit message for notes.

erikarvstedt commented:
Sorry, I've somehow missed this PR.

[Fixed branch + fixups](https://github.com/erikarvstedt/nix-bitcoin/commits/joinmarket-0.8.3-ea).

- Split topics into separate commits to ease reviewing
- Add some fixups

nixbitcoin commented:
Squashed with three changes

- Rebased on master
- Updated darkscience Tor onion address also in `joinmarket-ob-watcher`
- Targeted v0.0.49 for release
